http/client/src/main/java/org/wildfly/security/http/client/hpi/ClientConfigProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client.hpi;
+
+import org.wildfly.security.http.client.exception.ClientConfigException;
+
+import java.net.URI;
+
+/**
+ * Interface to enable configuration of authentication properties on the client side.
+ * @author kekumar@redhat.com
+ */
+public interface ClientConfigProvider {
```

Comment:
@keshav-725 Just a minor, since this is not a java provider, we should rename this class and HttpMechClientConfigProviderImpl class. We can also make these private